### PR TITLE
RST version of README.extern

### PR DESCRIPTION
**Pull request by mppf**
This PR is formatting only. I intend to make content changes
and further beautification changes in a separate PR.

### CODE REVIEW

File: doc/release/technotes/README.extern.rst [hunk 1]
```diff
@@ -37,26 +36,28 @@ All of the following sections rely on the same general approach:
    For example, if an external function foo() was defined in foo.h and
    foo.c, it could be added to the compilation using either:
 
+   ::
```
Comment (ben-albrecht) [inline doc/release/technotes/README.extern.rst]: These look fine as is, but you can also use `.. ::code-block sh` for syntax recognized shell commands.

File: doc/release/technotes/README.extern.rst [hunk 2]
```diff
@@ -85,11 +86,13 @@ To help with mapping types between C and Chapel, Chapel installations
 contain a standard module named 'SysCTypes' (located under
 $CHPL_HOME/modules/standard/gen/...).  This module defines a number of
```
Comment (ben-albrecht) [inline doc/release/technotes/README.extern.rst]: I'd do something like ``$CHPL_HOME/modules/standard/gen/`` - would we do a cross-reference to SysCTypes documentation later?

**Comment by ben-albrecht** [comment]
Looking good other than my minor comments.

File: doc/release/technotes/README.extern.rst (hunk 2 above)
Comment (mppf) [inline doc/release/technotes/README.extern.rst]: I want to make more substantial changes to this file (including fixing such cross-references) but want to make that Part 2 in another pull request.
